tools/lib/gitlint-rules/rules.py

```diff
@@ -0,0 +1,132 @@
+    'allows', 'allowing', 'allowed',
+    'amends', 'amending', 'amended',
+    'bumps', 'bumping', 'bumped',
+    'calculates', 'calculating', 'calculated',
+    'changes', 'changing', 'changed',
+    'cleans', 'cleaning', 'cleaned',
+    'commits', 'committing', 'committed',
+    'corrects', 'correcting', 'corrected',
+    'creates', 'creating', 'created',
+    'darkens', 'darkening', 'darkened',
+    'disables', 'disabling', 'disabled',
+    'displays', 'displaying', 'displayed',
+    'drys', 'drying', 'dryed',
+    'ends', 'ending', 'ended',
+    'enforces', 'enforcing', 'enforced',
+    'enqueues', 'enqueuing', 'enqueued',
+    'extracts', 'extracting', 'extracted',
+    'finishes', 'finishing', 'finished',
+    'fixes', 'fixing', 'fixed',
+    'formats', 'formatting', 'formatted',
+    'guards', 'guarding', 'guarded',
+    'handles', 'handling', 'handled',
+    'hides', 'hiding', 'hid',
+    'increases', 'increasing', 'increased',
+    'ignores', 'ignoring', 'ignored',
+    'implements', 'implementing', 'implemented',
+    'improves', 'improving', 'improved',
+    'keeps', 'keeping', 'kept',
+    'kills', 'killing', 'killed',
+    'makes', 'making', 'made',
+    'merges', 'merging', 'merged',
+    'moves', 'moving', 'moved',
+    'permits', 'permitting', 'permitted',
+    'prevents', 'preventing', 'prevented',
+    'pushes', 'pushing', 'pushed',
+    'rebases', 'rebasing', 'rebased',
+    'refactors', 'refactoring', 'refactored',
+    'removes', 'removing', 'removed',
+    'renames', 'renaming', 'renamed',
+    'reorders', 'reordering', 'reordered',
+    'requires', 'requiring', 'required',
+    'restores', 'restoring', 'restored',
+    'sends', 'sending', 'sent',
+    'sets', 'setting',
+    'separates', 'separating', 'separated',
+    'shows', 'showing', 'showed',
+    'skips', 'skipping', 'skipped',
+    'sorts', 'sorting',
+    'speeds', 'speeding', 'sped',
+    'starts', 'starting', 'started',
+    'supports', 'supporting', 'supported',
+    'takes', 'taking', 'took',
+    'testing', 'tested',  # 'tests' excluded to reduce false negative
+    'truncates', 'truncating', 'truncated',
+    'updates', 'updating', 'updated',
+    'uses', 'using', 'used'
+}
+
+imperative_forms = [
+    'add', 'allow', 'amend', 'bump', 'calculate', 'change', 'clean',
+    'commit', 'correct', 'create', 'darken', 'disable', 'display',
+    'dry', 'end', 'enforce', 'enqueue', 'extract', 'finish', 'fix',
+    'format', 'guard', 'handle', 'hide', 'increase', 'ignore',
+    'implement', 'improve', 'keep', 'kill', 'make', 'merge', 'move',
+    'permit', 'prevent', 'push', 'rebase', 'refactor', 'remove',
+    'rename', 'reorder', 'require', 'restore', 'send', 'set',
+    'separate', 'show', 'skip', 'sort', 'speed', 'start', 'support',
+    'take', 'test', 'truncate', 'update', 'use'
+]
+
+def head_binary_search(key, words):
+    # type: (Text, List[str]) -> str
+    """ Find the imperative tense by looking at the first 3 characters. """
```

Comment:
So this is very nitpicky and may not be worth us changing, especially in code we are drawing from an upstream project, but: imperative isn't a tense, it's a [mood](https://en.wikipedia.org/wiki/Imperative_mood). 

> [tense (location in time), aspect (fabric of time – a single block of time, continuous flow of time, or repetitive occurrence), and mood or modality (degree of necessity, obligation, probability, ability).](https://en.wikipedia.org/wiki/Tense%E2%80%93aspect%E2%80%93mood)

Giving you a heads-up in case this particular aspect of grammar pedantry moves you to change "tense" to "mood" in your PR. But I understand if it does not. :)